actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -2,8 +2,14 @@ module ActionController
   module ImplicitRender
     def send_action(method, *args)
       ret = super
-      default_render unless performed?
-      ret
+      return ret if performed?
+
+      if template_exists?(method, _prefixes, variants: request.variant)
```

Comment:
I can see it working there and you make good points about the public api.
I'll get on that.

On Sunday, April 5, 2015, Matthew Draper <notifications@github.com> wrote:

> In actionpack/lib/action_controller/metal/implicit_render.rb
> <https://github.com/rails/rails/pull/19377#discussion_r27779451>:
>
> > @@ -2,8 +2,14 @@ module ActionController
> >    module ImplicitRender
> >      def send_action(method, *args)
> >        ret = super
> > -      default_render unless performed?
> > -      ret
> > +      return ret if performed?
> > +
> > +      if template_exists?(method, _prefixes, variants: request.variant)
>
> I tried to follow the preceding comments, but I couldn't find why this
> isn't inside default_render. That's public API, and I think it's fair to
> consider the :no_content fallback to be part of the default rendering
> behaviour.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/19377/files#r27779451>.
>


-- 


  Steve Bussey •SalesLoft

•Software Engineer

e steve.bussey@salesloft.com w salesloft.com
<http://salesloft.com/?v=1&utm_expid=76406993-1.E5ncJhwyTAW9uGaR1WdBTA.1?utm_source=Email-Signature-Rescue&utm_medium=Email-Signature&utm_campaign=Email-Signature-Rescue>

a 3423 Piedmont Rd NE, Atlanta, GA 30305

[image: Facebook] <https://www.facebook.com/SalesLoft> [image: Twitter]
<https://twitter.com/SalesLoft> [image: Linkedin]
<https://www.linkedin.com/company/salesloft> [image: wordpress]
<http://blog.salesloft.com/blog/> [image:
http://salesloft.com/prospector/#video]
<http://salesloft.com/prospector/#video>  [image: Software Engineer Logo]
<http://salesloft.com/?v=1&utm_expid=76406993-1.E5ncJhwyTAW9uGaR1WdBTA.1?utm_source=Email-Signature-Rescue&utm_medium=Email-Signature&utm_campaign=Email-Signature-Rescue>